src/App.jsx

```diff
@@ -14,45 +17,111 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_ABC = 'abc';
+const SORT_FIELD_LENGTH = 'length';
+const RESET_FIELD = '';
+
+export const App = () => {
+  let visibleGoods = [...goodsFromServer];
+  const [reverse, setReverse] = useState(false);
+  const [sortField, setSortField] = useState('');
+  const hasResetButton = reverse || sortField;
+
+  const handleSort = field => () => setSortField(field);
+  const handleReverse = reversed => () => setReverse(!reversed);
+  const handleReset = (reverseReset, sortReset) => () => {
```

Comment:
Don't need to set reverseReset in for every time, when you execute handleReset. You can set just default value for this parameter.